### PR TITLE
CBBF-121 abstract common header fields

### PR DESCRIPTION
**Pull request by dho937**

### CODE REVIEW

File: bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java [hunk 1]
```diff
@@ -893,6 +894,65 @@ static void mapEobType(ExplanationOfBenefit eob, ClaimType blueButtonClaimType,
...
+		else
+			eob.addIdentifier().setSystem(TransformerConstants.CODING_CCW_CLAIM_ID).setValue(claimId);
+
+		eob.addIdentifier().setSystem(TransformerConstants.CODING_CCW_CLAIM_GROUP_ID)
+				.setValue(claimGroupId);
+
+		eob.getInsurance()
+				.setCoverage(referenceCoverage(beneficiaryId, coverageType));
+		eob.setPatient(referencePatient(beneficiaryId));
+		eob.setStatus(ExplanationOfBenefitStatus.ACTIVE);
```
Comment (karlmdavis) [inline bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java]: Ought to be mapped based on final action status. `F` for `ACTIVE`, `N` for `CANCELLED` sound right to you?

File: bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java [hunk 2]
```diff
@@ -893,6 +894,65 @@ static void mapEobType(ExplanationOfBenefit eob, ClaimType blueButtonClaimType,
...
+	 *            CLM_FROM_DT
+	 * @param dateThrough
+	 *            CLM_THRU_DT
+	 * @param paymentAmount
+	 *            CLM_PMT_AMT
+	 * @param finalAction
+	 *            FINAL_ACTION
+	 * 
+	 */
+	static void mapEobCommonClaimHeaderData(ExplanationOfBenefit eob, String claimId,
```
Comment (karlmdavis) [inline bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java]: Great method name. :+1:

File: bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java [hunk 3]
```diff
@@ -1778,9 +1833,12 @@ static void setProviderNumber(ExplanationOfBenefit eob, String providerNumber) {
 	 *            the {@link Optional}&lt;{@link Character}&gt;
 	 *            CARR_CLM_HCPCS_YR_CD: representing the hcpcs year code for the
 	 *            claim
+	 * @param hcpcsCode
+	 *            the {@link Optional}&lt;{@link String}&gt; HCPCS_CD: representing
+	 *            the hcpcs code for the claim
 	 */
 	static void setHcpcsModifierCodes(ItemComponent item, Optional<String> hcpcsInitialModifierCode,
-			Optional<String> hcpcsSecondModifierCode, Optional<Character> hcpcsYearCode) {
+			Optional<String> hcpcsSecondModifierCode, Optional<Character> hcpcsYearCode, Optional<String> hcpcsCode) {
```
Comment (karlmdavis) [inline bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java]: Do you think this method would make more sense if the params were reordered to put `hcpcsCode` first?

Eclipse has a "Change Method Signature" refactoring that will do that for you automagically, if you decide to.

File: bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java (hunk 1 above)
Comment (dho937) [reply bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java]: Yes, I agree.  Brian is going to make this change in CBBF-104 this week.  This is what Brian and I discussed last Friday.

File: bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java (hunk 3 above)
Comment (dho937) [reply bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java]: I agree that the hcpcsCode should be the first parameter.  Thanks for the tip on the "Change Method Signature" function within Eclipse...very slick feature.

File: bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java (hunk 1 above)
Comment (karlmdavis) [reply bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java]: 👍

**Pull request merged**